src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -216,16 +216,16 @@ public string RootId
         {
```

Comment:
Shouldn't the KeyValuePair be of `string?, string?` I don't see anything preventing people from adding null strings:
https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/Activity.cs,310

Reply:
I was assuming so, at lease value could be nullable but KeyValuePair constructor not accepting null for both, I also checked on corelib, constructor not nullable

Reply:
I think I'm missing something here, `KeyValuePair` clearly accepts null for both, the key and the value. We intentionally did that, as `KeyValuePair` can be used for a lot of different scenarios, https://source.dot.net/#System.Private.CoreLib/shared/System/Collections/Generic/KeyValuePair.cs,55
https://source.dot.net/#System.Private.CoreLib/shared/System/Collections/Generic/KeyValuePair.cs,22

Reply:
never mind, I see i have confused with generics nullable